app/gateway/2.7.x/reference/proxy.md

```diff
@@ -1325,6 +1325,33 @@ This means all of the below setup are supported in this mode:
 protocol list are supported. This list can be found in their respective documentation
 on [Kong Hub](https://docs.konghq.com/hub/).
 
+[Back to top](#introduction)
+
+## Proxy TLS passthrough traffic
+
+Starting with version 2.7, Kong supports proxying a TLS request without terminating or known
+as SNI proxy.
+
+Kong will not decrypt the incoming TLS traffic, but use the connecting client's
+TLS SNI server name extension to find the matching Route and Service and forward the
+complete TLS request to upstream.
+
+In this mode, request must be sent to a port that has `ssl` flag in its `stream_listen` directive.
+Aside from defining `stream_listen`, appropriate Route
+object with protocol types of `tls_passthrough` and `snis` field should be created, the corresponding
+Service object must have protocol of `tcp`. Different from proxying TLS traffic,
+SNI and Certificate entity is not required to be set and will not be used.
```

Comment:
```suggestion
In this mode, you need to:
* Create a Route object with the protocol `tls_passthrough`, and the 
`snis` field set to one or more SNIs.
* Set the corresponding Service object's protocol to `tcp`. 
* Send requests to a port that has the `ssl` flag in its `stream_listen` 
directive.

Separate SNI and Certificate entities aren't required and won't be used.
```

I moved the `stream_listen` point to the end of the list, as it seems like this is the order of operations - create Service & Route, then send requests.